src/main/java/seedu/address/logic/StatisticsManager.java

```diff
@@ -0,0 +1,97 @@
+package seedu.address.logic;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.Statistics;
+import seedu.address.model.dayData.DayData;
+
+import static seedu.address.model.dayData.CustomQueue.CONSTANT_SIZE;
+
+public class StatisticsManager {
+    private Statistics statistics;
+    private String dailyTargetText;
+    private String progressDailyText;
+    private String progressBarDailyFilepathString;
+    private ObservableList<DayData> customQueue;
+
+    public StatisticsManager() { }
+
+    public void setStatistics(Statistics statistics) {
+        this.statistics = statistics;
+    }
+
+    public void updateStatistics(ObservableList<DayData> customQueue) {
+        this.customQueue = customQueue;
+        DayData latestDayData = customQueue.get(CONSTANT_SIZE - 1);
+        int currProgress = latestDayData.getPomDurationData().value;
+
+        String currTargetText = statistics.getDailyTarget();
+        int currTarget = Integer.valueOf(currTargetText);
+        if (currProgress >= currTarget) {
```

Comment:
actually should we provide feedback that they've maxed it out

Reply:
sounds good! let's address it in another issue